algorithms.go

```diff
@@ -29,10 +29,8 @@ import (
 
 // Implements token bucket algorithm for rate limiting. https://en.wikipedia.org/wiki/Token_bucket
 func tokenBucket(ctx context.Context, s Store, c Cache, r *RateLimitReq) (resp *RateLimitResp, err error) {
-	ctx = tracing.StartScopeDebug(ctx)
-	defer func() {
-		tracing.EndScope(ctx, err)
-	}()
+	ctx = tracing.StartNamedScopeDebug(ctx, "tokenBucket")
+	defer tracing.EndScope(ctx, err)
```

Comment:
You must wrap `EndScope()` in an anonymous func or `err` will always be the initial value!  This needs to be addressed in all the other places this was done.